src/treeffuser/_score_models.py

```diff
@@ -132,19 +118,20 @@ def _make_training_data(
         val_mean, val_std = sde.get_mean_std_pt_given_y0(y_test, t_val)
         perturbed_y_val = val_mean + val_std * z_val
         predictors_val = np.concatenate(
-            [perturbed_y_val, X_test, t_val.reshape(-1, 1)], axis=1
+            [X_test, perturbed_y_val, t_val.reshape(-1, 1)], axis=1
         )
         predicted_val = -1.0 * z_val
 
-    return predictors_train, predictors_val, predicted_train, predicted_val
+    # cat_idx is not changed
```

Comment:
It happens that it is not changed, but if the concatenation was changing cat_idx (e.g. concatenating y left of x), then it would be super important to update cat_idx and return the changed version. I set that up so that in the future if we prefer to concatenate [y,x,t] we don't forget to update cat_idx